### PR TITLE
[#2861]Remove dependency

### PR DESCRIPTION
**Pull request by lin-mt**
Please do not create a Pull Request without creating an issue first.

## What is the purpose of the change

去掉 common-lang3 包

## Brief changelog

XX

## Verifying this change

XXXX

Follow this checklist to help us incorporate your contribution quickly and easily:

* [x] Make sure there is a Github issue filed for the change (usually before you start working on it). Trivial changes like typos do not require a Github issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue.
* [x] Format the pull request title like `[ISSUE #123] Fix UnknownException when host config not exist`. Each commit in the pull request should have a meaningful subject line and body.
* [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
* [ ] Write necessary unit-test to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add integration-test in [test module](https://github.com/alibaba/nacos/tree/master/test).
* [x] Run `mvn -B clean package apache-rat:check findbugs:findbugs -Dmaven.test.skip=true` to make sure basic checks pass. Run `mvn clean install -DskipITs` to make sure unit-test pass. Run `mvn clean test-compile failsafe:integration-test`  to make sure integration-test pass.

### CODE REVIEW

**Comment by chuntaojun** [comment]
please resolve conflict

**Comment by lin-mt** [comment]
> please resolve conflict

It's done.

**Comment by chuntaojun** [comment]
you need to resolve code conflict again

File: api/src/main/java/com/alibaba/nacos/api/NacosFactory.java [hunk 1]
```diff
@@ -19,7 +19,7 @@
 
 import com.alibaba.nacos.api.config.ConfigFactory;
 import com.alibaba.nacos.api.config.ConfigService;
-import com.alibaba.nacos.api.exception.NacosException;
+import com.alibaba.nacos.common.exception.api.NacosException;
```
Comment (chuntaojun) [inline api/src/main/java/com/alibaba/nacos/api/NacosFactory.java:22]: Exceptions are kept in the API module, and exceptions in the Common module are re-created

**Comment by zongtanghu** [comment]
我在这个pr(https://github.com/alibaba/nacos/pull/3031 )里面修改了客户端，去除客户端的对apache-common-lang3依赖，我看你这个应该跟我有一些重复哈，所以我想咱们是否可以一起合作下，你来移除server端对apache-common-lang3依赖. @lin-mt

**Comment by KomachiSion** [comment]
这个我涉及面我看很广，你们可以在issue中商量一下，把任务拆细，比如一个package一个PR，分5～6个PR改，这样及容易分配任务，又容易review

**Comment by zongtanghu** [comment]
> 这个涉及面我看很广，你们可以在issue中商量一下，把任务拆细，比如一个package一个PR，分5～6个PR改，这样及容易分配任务，又容易review @zongtanghu @lin-mt

是的，所以我上面贴的PR的链接，是实现了客户端那部分的，可以像上面两位说的，我们可以分工下哈，你主要来整server端的那部分，你看呢？

**Comment by lin-mt** [comment]
> > 这个涉及面我看很广，你们可以在issue中商量一下，把任务拆细，比如一个package一个PR，分5～6个PR改，这样及容易分配任务，又容易review @zongtanghu @lin-mt
> 
> 是的，所以我上面贴的PR的链接，是实现了客户端那部分的，可以像上面两位说的，我们可以分工下哈，你主要来整server端的那部分，你看呢？

可以的，你看下你除了commom、client和core之外还要修改其他的 package 吗，还是剩下的我来改，都可以的，你改完了我再把你的代码merge到我的分支，然后我再重新提交pr吧！这样可以吗

File: api/src/main/java/com/alibaba/nacos/api/NacosFactory.java (hunk 1 above)
Comment (lin-mt) [reply api/src/main/java/com/alibaba/nacos/api/NacosFactory.java:22]: 在common包中的com.alibaba.nacos.common.http.handler.ResponseHandler#responseEntityExtractor会抛出异常com.alibaba.nacos.api.exception.runtime.NacosDeserializationException，这个异常在包api中，而api包如果去掉 common-lang3 的话就需要依赖common，这就导致了循环依赖，是否可以将这个异常移动到common包中，如果有其他类似异常的话是否也这样处理

Comment (chuntaojun) [reply api/src/main/java/com/alibaba/nacos/api/NacosFactory.java:22]: common模块就使用Java内部有的异常吧

**Comment by KomachiSion** [comment]
@lin-mt Sorry, Can you split this PR to multiple PRs. It's hard to review because the PR have large range. Even we review finished, it may conflict with others small PR.

**Comment by KomachiSion** [comment]
I will open a new issue to discuss remove this dependency and split task.

**Comment by zongtanghu** [comment]
It's good to split to serveral small pr!



huzongtang@cmss.chinamobile.com
 
From: 杨翊 SionYang
Date: 2020-06-18 09:31
To: alibaba/nacos
CC: Hu Zongtang; Mention
Subject: Re: [alibaba/nacos] [#2861]Remove dependency (#2952)
I will open a new issue to discuss remove this dependency and split task.

**Comment by KomachiSion** [comment]
@lin-mt @zongtanghu see #3102

**Comment by lin-mt** [comment]
> @lin-mt @zongtanghu see #3102

OK

**Pull request closed without merging**